### DISCUSSION TITLE
Data format: JSON vs TAP vs XML

### DISCUSSION THREAD

**Discussion by JoshuaKGoldberg** [ota4j-team/open-test-reporting#223] [General]
👋 coming over from https://github.com/mochajs/mocha/issues/5271 to leave feedback as suggested (thanks for the reach-out!). Just surfacing for visibility, _if_ you want widespread adoption in the JavaScript ecosystem, I think there are two downsides to using XML:

* Most JS ecosystem developers & projects actively avoid using XML. Many of us prefer JSON for being lighter-weight and closer in semantics to the JavaScript language. Anything based on XML is starting at a perception disadvantage.
  * JSON is also much easier to describe in TypeScript-esque static types, which is a huge boon for JS/TS.
* The JS world already has a cross-framework standard: [TAP (Test Anything Protocol)](https://testanything.org). It's a well-defined spec that came out of Perl.
  * I'm not deeply familiar enough with TAP or OTR to give a full description of where there's overlap. It does look like there's a non-trivial amount.

It's understandable that XML is a common standard data format for Java area frameworks as well as other languages, so it makes sense that you'd use XML. I'm not requesting changes here. Just wanted to surface something relevant for JS-land. 🙂

**Comment by marcphilipp** [comment]
Thanks for providing context and interesting links! :+1:
